Code/GraphMol/FilterCatalog/ThreadedFilterCatalogRunner.h

```diff
@@ -0,0 +1,81 @@
+//  Copyright (c) 2022 Brian P Kelley
+//  All rights reserved.
+//
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#ifndef THREADED_FILTERCATALOG_RUNNER
+#define THREADED_FILTERCATALOG_RUNNER
+#include "FilterCatalog.h"
+#include "Filters.h"
+#include "FilterMatchers.h"
+#include <GraphMol/SmilesParse/SmilesParse.h>
+
+#ifdef RDK_THREADSAFE_SSS
+#include <RDGeneral/RDThreads.h>
+#include <thread>
+#include <future>
+#endif
+
+namespace RDKit {
+boost::shared_ptr<FilterCatalogEntry> &makeBadSmilesEntry() {
+  static boost::shared_ptr<FilterCatalogEntry> bad_smiles(
+      new FilterCatalogEntry("no valid RDKit molecule",
+                             boost::shared_ptr<FilterMatcherBase>()));
+  return bad_smiles;
+}
+template<class ThreadInitializer>
+void CatalogSearcher(
+    const FilterCatalog &fc, const std::vector<std::string> &smiles,
+    std::vector<std::vector<FilterCatalog::CONST_SENTRY>> &results, int start,
+    int numThreads) {
+  ThreadInitializer t;
```

Comment:
This is the real change here which required moving to a header file from the .cpp file